notebooks/fs.ipynb

```diff
@@ -0,0 +1,1340 @@
+      "\n",
+      "copy(config: Union[dict, str, pathlib.Path, NoneType] = None, target_dir: Union[str, pathlib.Path, NoneType] = None, cycle: Optional[datetime.datetime] = None, leadtime: Optional[datetime.timedelta] = None, keys: Optional[list[str]] = None, dry_run: bool = False, stdin_ok: bool = False) -> bool\n",
+      "    Copy files.\n",
+      "\n",
+      "    :param config: YAML-file path, or ``dict`` (read ``stdin`` if missing or ``None``).\n",
+      "    :param target_dir: Path to target directory.\n",
+      "    :param cycle: A datetime object to make available for use in the config.\n",
+      "    :param leadtime: A timedelta object to make available for use in the config.\n",
+      "    :param keys: YAML keys leading to file dst/src block.\n",
+      "    :param dry_run: Do not copy files.\n",
+      "    :param stdin_ok: OK to read from ``stdin``?\n",
+      "    :return: ``True`` if all copies were created.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(fs.copy)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0585971b-47c6-48aa-9f1f-d5890cbb2061",
+   "metadata": {},
+   "source": [
+    "Files to be copied are specified using a configuration YAML file, with source paths and names of the files to copy given as the values and the destination paths and names of copied files given as corresponding keys."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "a959522f-d769-48c6-918d-d42776b3600a",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "file1-copy.nml: fixtures/fs/file1.nml\n",
+      "data/file2-copy.txt: fixtures/fs/file2.txt\n",
+      "data/file3-copy.csv: fixtures/fs/file3.csv\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat fixtures/fs/copy-config.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5885fd76-78de-4814-ad7b-dfd6df18a07d",
+   "metadata": {},
+   "source": [
+    "With these instructions, `copy()` creates a copy of each given file with the given name and in the given subdirectory. Copies are created in the directory indicated by `target_dir`. Paths can be provided either as a string or <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> object. Any directories in the targeted paths for copying will be created if they don't already exist. `True` is returned upon a successful copy."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "a6aff6e3-815c-496e-81d7-d8756be9c232",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "fs.copy(\n",
+    "    config=\"fixtures/fs/copy-config.yaml\",\n",
+    "    target_dir=Path(\"tmp/copy-target\")\n",
+    ")"
+   ]
```

Comment:
```suggestion
   "source": [
    "rmtree(\"tmp\", ignore_errors=True)\n",
    "fs.copy(\n",
    "    config=\"fixtures/fs/copy-config.yaml\",\n",
    "    target_dir=Path(\"tmp/copy-target\")\n",
    ")"
   ]
```
Well, I would expect the addition of this `rmtree()` call to make this cell behave the same way no matter how many times the user runs it, but that doesn't seem to be the case. I'm confused by similar behavior in a later cell that I commented on and wonder if this may be related to caching. We should probably figure out what's going on here.